seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/rest/RestService.java

```diff
@@ -0,0 +1,674 @@
+        }
+        return identifier;
+    }
+
+    @VisibleForTesting
+    static String randomEndpoint(String feNodes, Logger logger) throws DorisConnectorException {
+        logger.trace("Parse fenodes '{}'.", feNodes);
+        if (StringUtils.isEmpty(feNodes)) {
+            String errMsg =
+                    String.format(ErrorMessages.ILLEGAL_ARGUMENT_MESSAGE, "fenodes", feNodes);
+            throw new DorisConnectorException(DorisConnectorErrorCode.REST_SERVICE_FAILED, errMsg);
+        }
+        List<String> nodes = Arrays.asList(feNodes.split(","));
+        Collections.shuffle(nodes);
+        return nodes.get(0).trim();
+    }
+
+    @VisibleForTesting
+    static List<String> allEndpoints(String feNodes, Logger logger) throws DorisConnectorException {
+        logger.trace("Parse fenodes '{}'.", feNodes);
+        if (StringUtils.isEmpty(feNodes)) {
+            String errMsg =
+                    String.format(ErrorMessages.ILLEGAL_ARGUMENT_MESSAGE, "fenodes", feNodes);
+            throw new DorisConnectorException(DorisConnectorErrorCode.REST_SERVICE_FAILED, errMsg);
+        }
+        List<String> nodes =
+                Arrays.stream(feNodes.split(",")).map(String::trim).collect(Collectors.toList());
+        Collections.shuffle(nodes);
+        return nodes;
+    }
+
+    @VisibleForTesting
+    public static String randomBackend(DorisConfig dorisConfig, Logger logger)
+            throws DorisConnectorException, IOException {
+        List<BackendV2.BackendRowV2> backends = getBackendsV2(dorisConfig, logger);
+        logger.trace("Parse beNodes '{}'.", backends);
+        if (backends == null || backends.isEmpty()) {
+            logger.error(ErrorMessages.ILLEGAL_ARGUMENT_MESSAGE, "beNodes", backends);
+            String errMsg =
+                    String.format(ErrorMessages.ILLEGAL_ARGUMENT_MESSAGE, "beNodes", backends);
+            throw new DorisConnectorException(DorisConnectorErrorCode.REST_SERVICE_FAILED, errMsg);
+        }
+        Collections.shuffle(backends);
+        BackendV2.BackendRowV2 backend = backends.get(0);
+        return backend.getIp() + ":" + backend.getHttpPort();
+    }
+
+    public static String getBackend(DorisConfig dorisConfig, Logger logger)
+            throws DorisConnectorException {
+        try {
+            return randomBackend(dorisConfig, logger);
+        } catch (Exception e) {
+            String errMsg = "Failed to get backend via " + dorisConfig.getFrontends();
+            throw new DorisConnectorException(
+                    DorisConnectorErrorCode.REST_SERVICE_FAILED, errMsg, e);
+        }
+    }
+
+    @Deprecated
+    @VisibleForTesting
+    static List<BackendRow> getBackends(DorisConfig dorisConfig, Logger logger)
+            throws DorisConnectorException, IOException {
+        String feNodes = dorisConfig.getFrontends();
+        String feNode = randomEndpoint(feNodes, logger);
+        String beUrl = "http://" + feNode + BACKENDS;
```

Comment:
What about abstracting away the url build?

Reply:
done